nixos/modules/system/activation/activatable-system.nix

```
@@ -60,6 +62,17 @@ in
         do, but for image based systems, this may not be needed or not be desirable.
       '';
     };
+    system.rebuildable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Exclude switch-to-confguration.pl from toplevel, rendering the system
+        unable to be re-built via `nixos-rebuild.
```

```suggestion
        Whether to include the capability to switch configurations without reboot.
        
        Disabling this makes the system unable to be reconfigured via `nixos-rebuild switch`.
```

- Describe what it does instead of why you added it
- I believe it could still be booted into a new configuration as in `nixos-rebuild boot` (e.g. a specialisation or hardware appliance updated with Nix)
- Missing backtick

---

For `nixos-rebuild boot` we would still need switch-to-configuration because that's what installs the bootloader (and thus the new entries pointing to the new init), right?

---

For the bootloader it delegates almost immediately to the `installBootLoader` script and then immediately exits, but yeah you're right.
Would be nice to factor that out. That would change the interface with `nixos-rebuild`, so should be done with care. Maybe later.